pkg/client/unversioned/request.go

```diff
@@ -642,6 +642,7 @@ func (r *Request) Watch() (watch.Interface, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
+		defer resp.Body.Close()
```

Comment:
no, otherwise the returned streamwatcher would be useless, since the underlying body would have been closed.